plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,609 @@
+  # insecure_skip_verify = true
+`
+}
+
+func (r *RedisSentinel) Description() string {
+	return "Read metrics from one or many redis-sentinel servers"
+}
+
+func (r *RedisSentinel) Init() error {
+	if len(r.Servers) == 0 {
+		r.Servers = []string{"tcp://localhost:26379"}
+	}
+
+	r.clients = make([]*RedisSentinelClient, len(r.Servers))
+
+	tlsConfig, err := r.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	for i, serv := range r.Servers {
+		u, err := url.Parse(serv)
+		if err != nil {
+			return fmt.Errorf("unable to parse to address %q: %s", serv, err.Error())
+		}
+
+		if u.Scheme != "tcp" && u.Scheme != "unix" {
+			return fmt.Errorf("invalid scheme %q. expected tcp or unix", u.Scheme)
+		}
+
+		password := ""
+
+		if len(r.Password) > 0 {
+			password = r.Password
+		}
+
+		var address string
+		if u.Scheme == "unix" {
+			address = u.Path
+		} else {
+			address = u.Host
+		}
+
+		sentinel := redis.NewSentinelClient(
+			&redis.Options{
+				Addr:      address,
+				Password:  password,
+				Network:   u.Scheme,
+				PoolSize:  1,
+				TLSConfig: tlsConfig,
+			},
+		)
+
+		tags := map[string]string{}
+		if u.Scheme == "unix" {
+			tags["socket"] = u.Path
+		} else {
+			tags["source"] = u.Hostname()
+			tags["port"] = u.Port()
+		}
+
+		r.clients[i] = &RedisSentinelClient{
+			sentinel: sentinel,
+			tags:     tags,
+		}
+	}
+
+	return nil
+}
+
+// Redis list format has string key/values adjacent, so convert to a map for easier use
+func toMap(vals []interface{}) map[string]string {
+	m := make(map[string]string)
+
+	for idx := 0; idx < len(vals)-1; idx += 2 {
+		key, keyOk := vals[idx].(string)
+		value, valueOk := vals[idx+1].(string)
+
+		if keyOk && valueOk {
+			m[key] = value
+		}
+	}
+
+	return m
+}
+
+// Reads stats from all configured servers accumulates stats.
+// Returns one of the errors encountered while gather stats (if any).
+func (r *RedisSentinel) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	for _, client := range r.clients {
+		wg.Add(1)
+
+		go func(client *RedisSentinelClient, acc telegraf.Accumulator) {
+			defer wg.Done()
+
+			gatherMasterStats(client, acc)
+			gatherInfoStats(client, acc)
+		}(client, acc)
+	}
+
+	wg.Wait()
+
+	return nil
+}
+
+func gatherInfoStats(client *RedisSentinelClient, acc telegraf.Accumulator) {
+	infoCmd := redis.NewStringCmd("info", "all")
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(infoCmd)
+
+	info, infoErr := infoCmd.Result()
+	if infoErr != nil {
+		acc.AddError(infoErr)
+		return
+	}
+
+	rdr := strings.NewReader(info)
+	infoTags, infoFields := convertSentinelInfoOutput(acc, client.BaseTags(), rdr)
+
+	acc.AddFields(measurementSentinel, infoFields, infoTags)
+}
+
+func gatherMasterStats(client *RedisSentinelClient, acc telegraf.Accumulator) {
+	mastersCmd := redis.NewSliceCmd("sentinel", "masters")
+	// We check the command result below
+	//nolint:errcheck
+	client.sentinel.Process(mastersCmd)
```

Comment:
Would it hurt to check the returned error here? Can you be sure that the result is a superset of errors returned by the function call?

Reply:
@srebhan The same error that gets returned here gets set on the command as well:

* https://github.com/go-redis/redis/blob/v6.15.9/redis.go#L200

Reply:
Well you now rely on an implementation detail in the external library. I would really feel more safe it we could check the error here too. If you cannot come up with a logic to avoid duplicate printing of the error I would rather accept duplicate lines in the log than missing out an error path...

Reply:
@srebhan Sure thing. Extra hardening can't hurt. Added in https://github.com/influxdata/telegraf/pull/10042/commits/93c5f7514b103e94ec08742f17e062f40e8962e9.